### PR TITLE
Update Build

### PR DESCRIPTION
**Pull request by isomarcte**
* Update various versions
* Add Scala3
* Add Scalafix/Scalafmt
    * Also add liancheng's ogranize imports scalafix tool
* Remove Main/MainSpec
* Switch to sbt-typelevel plugin

### CODE REVIEW

**Comment by isomarcte** [comment]
@rossabaker

File: build.sbt [hunk 1]
```diff
@@ -1,23 +1,16 @@
 import sbtcrossproject.CrossPlugin.autoImport.{crossProject, CrossType}
```
Comment (armanbilge) [inline build.sbt]:
```suggestion
```

Auto-imports are, auto-imported :)

File: version.sbt [hunk 2]
```diff
@@ -0,0 +1 @@
+ThisBuild / version := "0.0.1-SNAPSHOT"
```
Comment (armanbilge) [inline version.sbt]: sbt-typelevel has a git-based versioning scheme. You should add `ThisBuild / tlBaseVersion := "0.0"` to the primary build.
```suggestion
```

File: project/plugins.sbt [hunk 3]
```diff
@@ -1,2 +1,5 @@
-addSbtPlugin("io.github.davidgregory084" % "sbt-tpolecat" % "0.1.20")
-addSbtPlugin("io.chrisdavenport" % "sbt-davenverse" % "0.1.1")
\ No newline at end of file
+addSbtPlugin("ch.epfl.scala" % "sbt-scalafix" % "0.9.28")
+addSbtPlugin("com.github.cb372" % "sbt-explicit-dependencies" % "0.2.16")
+addSbtPlugin("com.timushev.sbt" % "sbt-updates" % "0.5.3")
+addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.4.2")
```
Comment (armanbilge) [inline project/plugins.sbt]: sbt-typelevel gets you the latest.
```suggestion
```

File: build.sbt [hunk 4]
```diff
@@ -1,23 +1,16 @@
...
-val munitCatsEffectV = "1.0.5"
-
-
 // Projects
-lazy val `cats-uri` = project.in(file("."))
+lazy val `cats-uri` = project
+  .in(file("."))
   .disablePlugins(MimaPlugin)
   .enablePlugins(NoPublishPlugin)
   .aggregate(core.jvm, core.js)
```
Comment (armanbilge) [inline build.sbt]:
```suggestion
lazy val root = tlCrossRootProject.aggregate(core)
```

File: build.sbt [hunk 5]
```diff
@@ -27,42 +20,12 @@ lazy val core = crossProject(JVMPlatform, JSPlatform)
...
-  .disablePlugins(MimaPlugin)
-  .enablePlugins(DavenverseMicrositePlugin)
-  .dependsOn(core.jvm)
-  .settings{
-    import microsites._
-    Seq(
-      micrositeDescription := "A URI with Cats instances",
-    )
-  }
+lazy val site = project.in(file("site")).disablePlugins(MimaPlugin).dependsOn(core.jvm)
```
Comment (armanbilge) [inline build.sbt]:
```suggestion
lazy val site = project.in(file("site")).enablePlugins(TypelevelSitePlugin).dependsOn(core.jvm)
```

You'll also need to add a dependency to `sbt-typelevel-site`.

**Review by rossabaker — COMMENTED** [review]
I didn't even remember there was a skeleton here.  You're in good hands with Arman.  I'll create a project team for it.

**Review by armanbilge — APPROVED** [review]

**Comment by rossabaker** [comment]
@armanbilge Are you hanging around to help with the project, or just here for the build? :smile:

**Comment by rossabaker** [comment]
Oops.  I did not look carefully at the build status before merging.

**Pull request merged**